resources/service_systemd.rb

```diff
@@ -0,0 +1,82 @@
+provides :nexus3_service_systemd
+
+provides :nexus3_service, os: 'linux' do |_node|
+  Chef::Platform::ServiceHelpers.service_resource_providers.include?(:systemd)
+end
+
+property :instance_name, String, name_property: true
+property :install_dir, String
+property :nexus3_user, String
+property :nexus3_group, String
+
+action :start do
+  create_init
+
+  service "nexus3_#{new_resource.instance_name}" do
+    provider Chef::Provider::Service::Systemd
+    supports restart: true, status: true
+    action :start
+    only_if 'command -v java >/dev/null 2>&1 || exit 1'
+  end
+end
+
+action :stop do
+  service "nexus3_#{new_resource.instance_name}" do
+    provider Chef::Provider::Service::Systemd
+    supports status: true
+    action :stop
+    only_if { ::File.exist?("/etc/systemd/system/nexus3_#{new_resource.instance_name}.service") }
+  end
+end
+
+action :restart do
+  service "nexus3_#{new_resource.instance_name}" do
+    provider Chef::Provider::Service::Systemd
+    supports status: true
+    action :restart
+  end
+end
+
+action :disable do
+  service "nexus3_#{new_resource.instance_name}" do
+    provider Chef::Provider::Service::Systemd
+    supports status: true
+    action :disable
+    only_if { ::File.exist?("/etc/systemd/system/nexus3_#{new_resource.instance_name}.service") }
+  end
+end
+
+action :enable do
+  create_init
+
+  service "nexus3_#{new_resource.instance_name}" do
+    provider Chef::Provider::Service::Systemd
+    supports status: true
+    action :enable
+    only_if { ::File.exist?("/etc/systemd/system/nexus3_#{new_resource.instance_name}.service") }
+  end
+end
+
+action_class do
+  def create_init
+    template "/etc/systemd/system/nexus3_#{new_resource.instance_name}.service" do
```

Comment:
This will be in another commit in this branch. Soon!

It is a bit convoluted to use version as the %i in systemd in the general case, but why not. I think we should probably use another identifier that people can set to their own liking.

As to using "service nexus3@3.2.1-01", this is messy on non-systemd systems.